src/app/routes/broker-ui/catalog-page/catalog-page/catalog-page.component.ts

```diff
@@ -98,6 +109,39 @@ export class CatalogPageComponent implements OnInit, OnDestroy {
       });
   }
 
+  private parseConnectorEndpointQueryParams() {
+    this.route.queryParams.pipe(take(1)).subscribe((x) => {
+      if (!('connectorEndpoint' in x)) {
+        return;
+      }
+      const endpoints = x.connectorEndpoint;
+      this.initialConnectorEndpointsFilterValues = Array.isArray(endpoints)
+        ? endpoints.filter((item, i, arr) => arr.indexOf(item) === i)
+        : [endpoints];
```

Comment:
my recommendation: make a method `parseConnectorEndpoints(params: Params): string[]` so you don't have to deal with this here